docs/manual/_sources/manual_circuit.rst.txt

```diff
@@ -951,21 +951,54 @@ See the `API reference <classical.html>`_ for more on the possible expressions a
     # if (reg_a >= 3)
     circ.T(qreg[7], condition=reg_geq(reg_a, 3))
     # compound register expressions
-    big_reg_exp = reg_a & reg_b | reg_c
+    big_reg_exp = (reg_a & reg_b) | reg_c
     circ.CX(qreg[3], qreg[4], condition=reg_eq(big_reg_exp, 3))
 
-.. Within the OpenQASM2 model, an entire register must be an exact value, the HQS extension enables more flexibility. OpenQASM3 will generalise even further.
+
+    # Write to classical registers
+    
+    # a = 3
+    circ.add_c_setreg(3, reg_a)
+    # a[0] = 1
+    circ.add_c_setbits([1], [reg_a[0]])
+    # Copy: b = a 
+    # b is smaller than a so the first 3 bits of a will be copied
+    circ.add_c_copyreg(reg_a, reg_b)
+    # b[1] = a[2]
+    circ.add_c_copybits([reg_a[2]], [reg_b[1]])
+    # c = a ^ b
+    circ.add_classicalexpbox_register(reg_a ^ reg_b, reg_c)
+    # c[0] = a[1] & b[2]
+    circ.add_classicalexpbox_bit(reg_a[1] & reg_b[2], [reg_c[0]])
+    # if (a > 1) b = 3
+    circ.add_c_setreg(3, reg_b, condition=reg_gt(reg_a, 1))
+
+    # Register arithmetic
+
+    # c = a + b // c (note the use of the floor divide symbol)
+    circ.add_classicalexpbox_register(reg_a + reg_b // reg_c, reg_c)
```

Comment:
I'm not clear on what it means for `add_classicalexpbox_register` to take multiple arguments? Is this new?

Reply:
the second argument is the register to write the result of the expression to